index.js

```diff
@@ -158,15 +158,16 @@ function monkeypatch() {
     }
   }
 
-  function checkIdentifierOrVisit(node) {
+  function checkIdentifierOrVisit(node, typeParameters) {
     if (node.typeAnnotation) {
       visitTypeAnnotation.call(this, node.typeAnnotation);
     } else if (node.type === "Identifier") {
       // exception for polymorphic types: <T>, <A>, etc
-      if (node.name.length === 1 && node.name === node.name.toUpperCase()) {
-        return;
+      if (node.name.length === 1 && node.name === node.name.toUpperCase() && typeParameters) {
```

Comment:
Do you still need to check if it's a one letter capital after this change? Not all type parameters are one letter capitals, so it would be nice if this new `typeParameters` bool sufficed :)